### DISCUSSION TITLE
Es posible que un required survey solo pueda ser completado por un perfil de usuario?

### DISCUSSION THREAD

**Discussion by jdurandpaz** [Cotalker/documentation#53] [2. Questions and Answers] [answered]
Hola,

Hice una máquina de estados simple, en la que configure required survey para la creación y cambio de estados. Mi duda es si hay alguna forma de evitar que ciertos usuarios puedan completar el formulario desde la vista de task, ya que agregué un permiso al survey pero solo funciona desde la vista de canal( donde deja de aparecer), pero en la vista de task sigue apareciendo como se ve en la imagenes adjuntas:

**VISTA CANAL** 

![image](https://user-images.githubusercontent.com/50208672/114087881-6275f480-9882-11eb-950e-cb66e06a8951.png)

**VISTA TASK** 

![image](https://user-images.githubusercontent.com/50208672/114087711-35c1dd00-9882-11eb-9e04-41ae8c8ee947.png)

Saludos

**Comment by acedward** [comment]
@jdurandpaz Como nota, algunas mejoras para este problema están en el documento interno "**Admin 2021 (Partner Edition)**" Punto **26**
Entiendo que la configuración del TaskGroup deberíá permitir esto, pero no estoy seguro.

@rafaelwalter podría saber más de este punto

**Comment by rafaelwalter** [comment, marked as answer]
@jdurandpaz actualmente solo tenemos una forma de filtrar usuarios que crean tasks.

Se aplica a nivel de TaskGroup y esto quiere decir que los usuarios que se filtren ahí no podran hacer nada, solamente ver las tasks del TaskGroup.

Aún no es posible configurar este filtro en el Admin, por lo que tendrán que configurar por API.

La endpoint para esto es `PATCH /api/v2/tasks/group/:taskGroupId`

El atributo a modificar es `taskGroup.writePermissions`. Este acepta un arreglo de strings que son los permissions de Cotalker.
Los usuarios que tienen al menos uno de los permisos definidos en `writePermissions` podrán hacer cualquier operacion de escritura en tasks (crear, actualizar, cambios de estado, etc).

**Reply by jdurandpaz** [reply]
@rafaelwalter gracias por la info, pero no se si aplica para mi caso, donde si quiero que un usuario pueda crear tasks (solicitudes) pero no quiero que pueda hacer un cambio de estado en particular que esta pensando para un supervisor. Entendería entonces que para evitar eso tendría que cuando llegue a ese estado dejarlo como solo visualizador o algo por el estilo y así evitar que pueda hacer cambios por la vista de task.

**Reply by acedward** [reply]
Efectivamente solamente los Editores pueden cambiar el estado manualmente. 
Si agregas el creador como follower - puede ver pero no hacer cambios directos (solo a través de formularios que tiene permisos para levantar)
